test/e2e/cypress/e2e/settings.cy.js

```diff
@@ -806,4 +807,73 @@ context('Settings page', () => {
       });
     });
   });
+  describe('Forbidden action', () => {
+    const password = 'password';
+
+    beforeEach(() => {
+      cy.deleteAllUsers();
+      cy.logout();
+      const user = createUserRequestFactory.build({
+        password,
+        password_confirmation: password,
+      });
+      cy.wrap(user).as('user');
+    });
+
+    it('should enable settings buttons if the user has the correct abilities', () => {
+      const userAbilites = [
+        { name: 'all', resource: 'activity_logs_settings' },
+        { name: 'all', resource: 'api_key_settings' },
+        { name: 'all', resource: 'suma_settings' },
+      ];
+      cy.get('@user').then((user) => {
+        cy.createUserWithAbilities(user, userAbilites);
+        cy.login(user.username, password);
+      });
+      cy.visit(`/settings`);
+      // API Key settings button
+      cy.contains('button', 'Generate Key')
+        .should('be.visible')
+        .and('be.enabled');
+      // SUSE Manager config settings button
+      cy.contains('button', 'Test Connection').should('be.enabled');
+      cy.get(
```

Comment:
Maybe there is a nice way to find the buttons, but well hehe

Reply:
Yeah, I know it is ugly, that's why I put some comments. 
As we have two "Edit Settings" buttons, I thought taking the selector directly with cypress to be precise won't harm :) 

Btw just an idea, we could add an aria label prop for the button component, with this we could have a very readable way to find elements like the two "Edit Settings" buttons

Reply:
Maybe you could find the by the h2 header. Something like: (not tested):
```
cy.contains('h2', 'SUSE Manager Config').next().contains('button', 'Edit Settings').should('be.enabled'):
```
Give a try if you want. Anyway, if the selectors find what they need to find, the tests does what it is supposed to do, so I'm fine as well

Reply:
Yeah, I was playing with cypress [eq](https://docs.cypress.io/api/commands/eq), [first ](https://docs.cypress.io/api/commands/first) and [last ](https://docs.cypress.io/api/commands/last) but it could not find the second edit settings button so meh. 
Your suggestion is much more readable, I would stick with it :+1: 
Thanks for the idea!